operations/configuration_management/Management_Node_Image_Customization.md

```diff
@@ -358,122 +358,111 @@ new customized image.
+1. (`ncn-mw#`) Determine the component names (xnames) for the management nodes which will boot from the new image.
 
-        > In this example, the xname for `ncn-w001` is being found.
+   > In this example, the xname for `ncn-w001` is being found.
 
-        ```bash
-        ssh ncn-w001 cat /etc/cray/xname
-        ```
+   ```bash
+   ssh ncn-w001 cat /etc/cray/xname
+   ```
 
 1. (`ncn-mw#`) Update boot parameters for a management node.
 
-    Perform the following procedure **for each xname** identified in the previous step.
-
     1. Get the existing `metal.server` setting for the component name (xname) of the node of interest.
 
-        ```bash
-        XNAME=<node-xname>
-        METAL_SERVER=$(cray bss bootparameters list --hosts "${XNAME}" --format json | jq '.[] |."params"' \
-            | awk -F 'metal.server=' '{print $2}' \
-            | awk -F ' ' '{print $1}')
-        echo "${METAL_SERVER}"
-        ```
+       ```bash
+       XNAME=<node-xname>
+       METAL_SERVER=$(cray bss bootparameters list --hosts "${XNAME}" --format json | jq '.[] |."params"' \
+           | awk -F 'metal.server=' '{print $2}' \
+           | awk -F ' ' '{print $1}')
+       echo "${METAL_SERVER}"
+       ```
 
     1. Create updated boot parameters that point to the new artifacts.
 
-        1. Set the path to the artifacts in S3.
+       1. Set the path to the artifacts in S3.
 
-            **NOTE:** This uses the `NEW_IMS_IMAGE_ID` variable set in an earlier step.
+          ***NOTE*** This uses the `NEW_IMS_IMAGE_ID` variable set in an earlier step.
 
-            ```bash
-            S3_ARTIFACT_PATH="boot-images/${NEW_IMS_IMAGE_ID}"
-            echo "${S3_ARTIFACT_PATH}"
-            ```
+          ```bash
+          S3_ARTIFACT_PATH="boot-images/${NEW_IMS_IMAGE_ID}"
+          echo "${S3_ARTIFACT_PATH}"
+          ```
 
-        1. Set the new `metal.server` value.
+       1. Set the new `metal.server` value.
 
-            ```bash
-            NEW_METAL_SERVER="s3://${S3_ARTIFACT_PATH}/rootfs"
-            echo "${NEW_METAL_SERVER}"
-            ```
+          ```bash
+          NEW_METAL_SERVER="s3://${S3_ARTIFACT_PATH}/rootfs"
+          echo "${NEW_METAL_SERVER}"
+          ```
 
-        1. Determine the modified boot parameters for the node.
+       1. Determine the modified boot parameters for the node.
 
-            ```bash
-            PARAMS=$(cray bss bootparameters list --hosts "${XNAME}" --format json | jq '.[] |."params"' | \
-                sed "/metal.server/ s|${METAL_SERVER}|${NEW_METAL_SERVER}|" | \
-                sed "s/metal.no-wipe=1/metal.no-wipe=0/" | \
-                tr -d \")
-            echo "${PARAMS}"
-            ```
+          ```bash
+          PARAMS=$(cray bss bootparameters list --hosts "${XNAME}" --format json | jq '.[] |."params"' | \
+              sed "/metal.server/ s|${METAL_SERVER}|${NEW_METAL_SERVER}|" | \
+              sed "s/metal.no-wipe=1/metal.no-wipe=0/" | \
```

Comment:
That section is for all NCNs. I see that the 1.4.4 patch use that section. I do think it should be removed since metal.no-wipe in each node's rebuild/upgrade workflow. However, it probably won't cause a problem to leave. I do think you need to remove the metal.no-wipe in the assign-ncn-images script because we don't want to set it on all nodes, especially storage nodes.

Reply:
I think it might need to be left in, because if someone is following this guide then they aren't using the workflow.

As for the script, I think you found a good bug. That script shouldn't be setting that, it is because I copied it from the docs. That script should only be updating the image, initrd, and kernel URLs.

That means I'll need a new step where users set `metal.no-wipe` for this upgrade from 1.4.x to 1.4.4.